### PR TITLE
Add basic edge property support for grpc

### PR DESCRIPTION
**Pull request by farodin91**
Part of #1853

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### CODE REVIEW

File: janusgraph-grpc/src/main/proto/janusgraph/types/v1/schema_types.proto [hunk 1]
```diff
@@ -84,6 +84,15 @@ message VertexLabel {
     repeated VertexProperty properties = 5;
 }
 
+message EdgeProperty {
+    // The Internal id.
+    google.protobuf.Int64Value id = 1;
+    // The VertexProperty Name.
```
Comment (FlorianHockmann) [inline janusgraph-grpc/src/main/proto/janusgraph/types/v1/schema_types.proto:90]: _edge_, not _vertex_

File: janusgraph-grpc/src/main/proto/janusgraph/types/v1/schema_types.proto [hunk 2]
```diff
@@ -116,4 +125,6 @@ message EdgeLabel {
     }
     // Defines Multiplicity of a edge.
     Multiplicity multiplicity = 4;
+    // Contains all constrained properties at the moment.
```
Comment (FlorianHockmann) [inline janusgraph-grpc/src/main/proto/janusgraph/types/v1/schema_types.proto:128]:
```suggestion
    // All properties constrained to this edge label.
```

File: janusgraph-server/src/test/java/org/janusgraph/graphdb/grpc/schema/SchemaManagerClientTest.java [hunk 3]
```diff
@@ -347,4 +348,74 @@ public void testGetEdgeLabels(int numberOfEdges) {
...
+
+    @ParameterizedTest
+    @EnumSource(value = PropertyDataType.class, mode = EXCLUDE, names = { "PROPERTY_DATA_TYPE_UNSPECIFIED", "UNRECOGNIZED" })
+    public void testGetEdgeLabelByNameEdgeLabelWithEdgeProperty(PropertyDataType propertyDataType) {
+        final String name = "testEdgePropertyType";
+        SchemaManagerClient schemaManagerClient = new SchemaManagerClient(getDefaultContext(), managedChannel);
+
+        //create property
+        EdgeProperty test = EdgeProperty.newBuilder()
+            .setName(name + "-prop")
```
Comment (FlorianHockmann) [inline janusgraph-server/src/test/java/org/janusgraph/graphdb/grpc/schema/SchemaManagerClientTest.java:384]: (nitpick) Just something for better readability: I would save this name in a variable which can then also be used in the assert in line 396.

**Review by FlorianHockmann — COMMENTED** [review]
LGTM in general, I just added some nitpick comments.

**Comment by farodin91** [comment]
@FlorianHockmann Rebased and all changes used.

File: janusgraph-grpc/src/main/proto/janusgraph/types/v1/schema_types.proto [hunk 4]
```diff
@@ -84,6 +84,15 @@ message VertexLabel {
     repeated VertexProperty properties = 5;
 }
 
+message EdgeProperty {
+    // The Internal id.
+    google.protobuf.Int64Value id = 1;
```
Comment (porunov) [inline janusgraph-grpc/src/main/proto/janusgraph/types/v1/schema_types.proto:89]: What's the difference with `int64` data type?

Comment (farodin91) [reply janusgraph-grpc/src/main/proto/janusgraph/types/v1/schema_types.proto:89]: If I'm correct, it allows the value to be nullable.

Comment (porunov) [reply janusgraph-grpc/src/main/proto/janusgraph/types/v1/schema_types.proto:89]: There is a keyword `optional` which allows to not provide any value for a scalar type.
I.e.:
```
optional int64 id = 1;
```
I just don't know if there is any reason to use `google.protobuf.Int64Value` if we have native support for 64 bit values

Comment (farodin91) [reply janusgraph-grpc/src/main/proto/janusgraph/types/v1/schema_types.proto:89]: Before 3.15, Protobuf 3 hasn't support the keyword optional.

**Review by FlorianHockmann — APPROVED** [review]

**Review by porunov — APPROVED** [review]
LGTM. Thank you @farodin91 !
The only thing I would like to use `optional int64 id = 1;` instead of `google.protobuf.Int64Value id = 1;` but as `google.protobuf.Int64Value` is used already in many places I think it makes sense to proceed with this PR and then open a new PR which changes `google.protobuf.Int64Value` to `optional int64` in all places at once.

File: janusgraph-grpc/src/main/proto/janusgraph/types/v1/schema_types.proto (hunk 4 above)
Comment (farodin91) [reply janusgraph-grpc/src/main/proto/janusgraph/types/v1/schema_types.proto:89]: I will do this refactoring in different PR.

**Comment by farodin91** [comment]
@porunov & @li-boxuan I fixed the id set behavior. I will refactor the id behavior for all cases in different PR.

**Comment by farodin91** [comment]
@porunov & @li-boxuan  Follow up PR: https://github.com/JanusGraph/janusgraph/pull/3020

**Review by li-boxuan — APPROVED** [review]
👍  LGTM

**Review by porunov — APPROVED** [review]
LGTM. Thank you @farodin91 !

**Comment by farodin91** [comment]
Merge it without passing the dist test. It is already failing on master. I will try to provide a pr asap.

**Comment by porunov** [comment]
Thank you @farodin91 . The issue regarding dist test is: #3017

**Pull request merged**